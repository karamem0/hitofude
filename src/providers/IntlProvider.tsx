//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
