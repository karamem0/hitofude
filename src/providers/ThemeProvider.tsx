//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
