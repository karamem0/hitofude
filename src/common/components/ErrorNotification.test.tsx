//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../providers/IntlProvider';
import Presenter from './ErrorNotification.presenter';
import ThemeProvider from '../../providers/ThemeProvider';
import { render } from '@testing-library/react';

it('should create shapshot', async () => {
  const params = {
    message: {
      id: 'ErrorMessage',
      defaultMessage: 'Something went wrong'
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
