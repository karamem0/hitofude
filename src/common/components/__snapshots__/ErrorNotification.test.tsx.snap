// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the message is not undedined 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  z-index: 900;
  width: calc(100% - 1rem);
  margin: 0.5rem;
  background-color: #fafafa;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      aria-labelledby="fui-r1"
      class="fui-MessageBar r2oyxsj emotion-0 ___hd2t6r0_1l4k1ej f13ftzij frd1ypx f1gyjrma fqyqtrt f18qd5xz"
      role="group"
    >
      <div
        class="fui-MessageBar__icon r1df1z33 ___gwngeu0_j951p00 f14a4cve"
      >
        <svg
          aria-hidden="true"
          class="___9ctc0p0_1xvj9ao f1w7gpdv fez10in f1dd5bof"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.68 2.79a1.5 1.5 0 0 1 2.64 0l6.5 12A1.5 1.5 0 0 1 16.5 17h-13a1.5 1.5 0 0 1-1.32-2.21l6.5-12ZM10.5 7.5a.5.5 0 0 0-1 0v4a.5.5 0 0 0 1 0v-4Zm.25 6.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="fui-MessageBarBody rtatq2b"
      >
        Something went wrong
      </div>
      <div
        class="fui-MessageBarActions r1t4x98y ___12vb27g_aqou110 fjseox"
      />
      <div
        class="fui-MessageBarActions__containerAction rgzw8nq"
      >
        <button
          class="fui-Link ___52mpgl0_k1g75g0 f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fkhj508 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f1iuv45f f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1ern45e f1n71otn f1h8hb77 f1deefiw"
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the message is undedined 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr2 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  />
</DocumentFragment>
`;
